btk/galsim_draw.py

```diff
@@ -0,0 +1,223 @@
+    dir: dictionary
+        dictionary that contains the information for the high resolution survey
+    shape_lr: tuple of ints
+        shape of the lr image
+    npsf: int
+        size on-a-side of a psf (in pixels)
+    cat: list
+        catalog where to draw galaxies from
+
+    Returns
+    -------
+    im_hr: galsim Image
+        galsim Image object with the high resolution simulated image and its WCS
+    im_lr: galsim Image
+        galsim Image object with the low resolution simulated image and its WCS
+    psf_hr: numpy array
+        psf of the high resolution image
+    psf_lr: numpy array
+        psf of the low resolution image
+    """
+    pix = dir["pixel"]
+    sigma = dir["psf"]
+
+    # Rotation angle
+    theta = np.random.randn(1) * np.pi * 0
+    angle = galsim.Angle(theta, galsim.radians)
+
+    # Image frames
+    im = galsim.Image(shape[0], shape[1], scale=pix)
+
+    # Galaxy profile
+    gal = cat.makeGalaxy(k, gal_type=gal_type, noise_pad_size=shape[0] * pix)
+    gal = gal.shift(dx=shift[0], dy=shift[1])
+    ## PSF is a Moffat profile dilated to the sigma of the corresponding survey
+    psf_int = galsim.Moffat(2, HST["pixel"]).dilate(sigma / HST["psf"]).withFlux(1.0)
+    ## Draw PSF
+    psf = psf_int.drawImage(
+        nx=npsf, ny=npsf, method="real_space", use_true_center=True, scale=pix_hr
+    ).array
+    ## Make sure PSF vanishes on the edges of a patch that has the shape of the initial npsf
+    psf = psf - psf[0, int(npsf / 2)] * 2
+    psf[psf < 0] = 0
+    psf = psf / np.sum(psf)
+    ## Interpolate the new 0-ed psf
+    psf_int = galsim.InterpolatedImage(galsim.Image(psf), scale=pix).withFlux(1.0)
+    ## Re-draw it (with the correct fulx)
+    psf = psf_int.drawImage(
+        nx=npsf, ny=npsf, method="real_space", use_true_center=True, scale=pix_hr
+    ).array
+
+    # Convolve galaxy profile by PSF, rotate and sample at high resolution
+    im = galsim.Convolve(gal, psf_int).drawImage(
+        nx=shape[0],
+        ny=shape[1],
+        use_true_center=True,
+        method="no_pixel",
+        scale=pix,
+        dtype=np.float64,
+    )
+    return im
+
+
+def mk_scene(hr_dict, lr_dict, cat, shape_hr, shape_lr, n_gal, gal_type):
+    """Generates blended scenes at two resolutions
+
+    Parameters
+    ----------
+    hr_dict, lr_dict: dictionaries
+        two dictionaries that contain the information for surveys (pixel scale, name and psf size)
+    cat: catalog
+        catalog of sources for galsim
+    shape_hr, shape_lr: tuples
+        2D shapes of the desired images for surveys indicated in the dictionaries
+    ngal: int
+        number of galaxies to draw on the scene
+    """
+    pix_hr = hr_dict["pixel"]
+
+    lr = 0
+    hr = 0
+    loc = []
+    for i in range(n_gal):
+        k = np.int(np.random.rand(1) * len(cat))
+        shift = (np.random.rand(2) - 0.5) * shape_hr * pix_hr / 2
+        ihr, ilr, phr, plr, _ = mk_sim(
```

Comment:
Does `mk_sim` in this line call the one in line 10? It seems the number of arguments is not consistent.